server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DLPConfigurationRoutesTest.java

```diff
@@ -0,0 +1,571 @@
+                    "]}");
+        }
+
+        @Test
+        void putShouldStoreTheConfigurationsWhenExplanationNotSpecified() {
+            String storeBody =
+                "{\"rules\": [{" +
+                "  \"id\": \"3\"," +
+                "  \"expression\": \"expression 3\"" +
+                "}]}";
+
+            given()
+                .body(storeBody)
+            .when()
+                .put(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.NO_CONTENT_204);
+
+            String retrievedBody = with()
+                .get(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .extract()
+                .body().asString();
+
+            assertThatJson(retrievedBody)
+                .isEqualTo(
+                    "{\"rules\": [" +
+                    "  {" +
+                    "    \"id\": \"3\"," +
+                    "    \"expression\": \"expression 3\"," +
+                    "    \"explanation\": null," +
+                    "    \"targetsSender\": false," +
+                    "    \"targetsRecipients\": false," +
+                    "    \"targetsContent\": false" +
+                    "  }" +
+                    "]}");
+        }
+
+        @Test
+        void putShouldReturnBadRequestWhenIdIsNotSpecified() {
+            String body =
+                "{" +
+                "  \"expression\": \"expression 4\"," +
+                "  \"explanation\": \"explanation 4\"," +
+                "  \"targetsSender\": false," +
+                "  \"targetsRecipients\": false," +
+                "  \"targetsContent\": false" +
+                "}";
+
+            given()
+                .body(body)
+            .when()
+                .put(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(400))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("JSON payload of the request is not valid"));
+        }
+
+        @Test
+        void putShouldReturnBadRequestWhenExpressionIsNotSpecified() {
+            String body =
+                "{" +
+                "  \"id\": \"5\"," +
+                "  \"explanation\": \"explanation 5\"," +
+                "  \"targetsSender\": false," +
+                "  \"targetsRecipients\": false," +
+                "  \"targetsContent\": false" +
+                "}";
+
+            given()
+                .body(body)
+            .when()
+                .put(DEFAULT_DOMAIN)
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(JSON_CONTENT_TYPE)
+                .body("statusCode", is(400))
+                .body("type", is("InvalidArgument"))
+                .body("message", is("JSON payload of the request is not valid"));
```

Comment:
Idem, I'd like the tests to ensure the cause reports the JSON is not valid **because** the expression is not defined...